### PR TITLE
kdlfmt: init at 0.0.3

### PR DESCRIPTION
**Pull request by airRnot1106**
A formatter for kdl documents
https://github.com/hougesen/kdlfmt

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -743,6 +743,11 @@
     githubId = 37664775;
     name = "Yuto Oguchi";
   };
+  airrnot = {
+    name = "airRnot";
+    github = "airRnot1106";
+    githubId = "62370527";
```
Comment (ryand56) [inline maintainers/maintainer-list.nix:749]:
```suggestion
    githubId = 62370527;
```

File: pkgs/by-name/kd/kdlfmt/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "v${version}";
+    hash = "sha256-qD1NYLHGmVRgV6pPXbvJ9NWDg/wVLWJY4hUsOLDlKh0=";
+  };
+
+  cargoHash = "sha256-7HSDz/JI5VuTdM/Hv+nq+ddpQg31Q1v7Ct5gz2PfdmE=";
+
+  meta = {
+    description = "A formatter for kdl documents";
+    homepage = "https://github.com/hougesen/kdlfmt.git";
+    changelog = "https://github.com/hougesen/kdlfmt/blob/${src.rev}/CHANGELOG.md";
```
Comment (ryand56) [inline pkgs/by-name/kd/kdlfmt/package.nix:23]:
```suggestion
    changelog = "https://github.com/hougesen/kdlfmt/blob/v${version}/CHANGELOG.md";
```

**Review by ryand56 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/kd/kdlfmt/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "hougesen";
+    repo = "kdlfmt";
+    rev = "v${version}";
+    hash = "sha256-qD1NYLHGmVRgV6pPXbvJ9NWDg/wVLWJY4hUsOLDlKh0=";
+  };
+
+  cargoHash = "sha256-7HSDz/JI5VuTdM/Hv+nq+ddpQg31Q1v7Ct5gz2PfdmE=";
+
+  meta = {
+    description = "A formatter for kdl documents";
```
Comment (Aleksanaa) [inline pkgs/by-name/kd/kdlfmt/package.nix:21]:
```suggestion
    description = "Formatter for kdl documents";
```

**Comment by airRnot1106** [comment]
Thanks for the suggestions. I fixed it.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**